pkg/kotsadm/objects/affinity_objects.go

```diff
@@ -1,34 +1,66 @@
 package kotsadm
 
 import (
+	"runtime"
+
 	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/selection"
 )
 
 func defaultKOTSNodeAffinity() *corev1.NodeAffinity {
+	darwinSelectorTerm := corev1.NodeSelectorTerm{
+		MatchExpressions: []corev1.NodeSelectorRequirement{
+			{
+				Key:      "kubernetes.io/os",
+				Operator: corev1.NodeSelectorOpIn,
+				Values: []string{
+					"linux",
+				},
+			},
+			{
+				Key:      "kubernetes.io/arch",
+				Operator: corev1.NodeSelectorOpIn,
+				Values: []string{
+					"arm64",
+					"amd64",
+				},
+			},
+		},
+	}
+
+	nonDarwinSelectorTerm := corev1.NodeSelectorTerm{
+		MatchExpressions: []corev1.NodeSelectorRequirement{
+			{
+				Key:      "kubernetes.io/os",
+				Operator: corev1.NodeSelectorOpIn,
+				Values: []string{
+					"linux",
+				},
+			},
+			{
+				Key:      "kubernetes.io/arch",
+				Operator: corev1.NodeSelectorOpNotIn,
+				Values: []string{
+					"arm64",
+				},
+			},
+		},
+	}
+
+	var selectorTerm corev1.NodeSelectorTerm
+	switch runtime.GOOS {
```

Comment:
This is evaluated in CLI context, correct?  So deployment is done into a Linux cluster from a Mac client, this will get the `darwinSelectorTerm`?

Reply:
that is correct, and that is what i believed to be the ask here